storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -0,0 +1,3640 @@
+            Map<List<Long>, List<Object>> execToNodePort = new HashMap<>();
+            for (Entry<ExecutorDetails, WorkerSlot> execAndNodePort: schedEntry.getValue().getExecutorToSlot().entrySet()) {
+                ExecutorDetails exec = execAndNodePort.getKey();
+                WorkerSlot slot = execAndNodePort.getValue();
+                
+                List<Long> listExec = new ArrayList<>(2);
+                listExec.add((long) exec.getStartTask());
+                listExec.add((long) exec.getEndTask());
+                
+                List<Object> nodePort = new ArrayList<>(2);
+                nodePort.add(slot.getNodeId());
+                nodePort.add((long)slot.getPort());
+                
+                execToNodePort.put(listExec, nodePort);
+            }
+            ret.put(schedEntry.getKey(), execToNodePort);
+        }
+        return ret;
+    }
+    
+    private static int numUsedWorkers(SchedulerAssignment assignment) {
+        if (assignment == null) {
+            return 0;
+        }
+        return assignment.getSlots().size();
+    }
+    
+    /**
+     * convert {topology-id -> SchedulerAssignment} to
+     *         {topology-id -> {[node port] [mem-on-heap mem-off-heap cpu]}}
+     * Make sure this can deal with other non-RAS schedulers
+     * later we may further support map-for-any-resources
+     * @param schedAssignments the assignments
+     * @return  {topology-id {[node port] [mem-on-heap mem-off-heap cpu]}}
+     */
+    private static Map<String, Map<List<Object>, List<Double>>> computeTopoToNodePortToResources(Map<String, SchedulerAssignment> schedAssignments) {
+        Map<String, Map<List<Object>, List<Double>>> ret = new HashMap<>();
+        for (Entry<String, SchedulerAssignment> schedEntry: schedAssignments.entrySet()) {
+            Map<List<Object>, List<Double>> nodePortToResources = new HashMap<>();
+            for (WorkerSlot slot: schedEntry.getValue().getExecutorToSlot().values()) {
+                List<Object> nodePort = new ArrayList<>(2);
+                nodePort.add(slot.getNodeId());
+                nodePort.add((long)slot.getPort());
+                
+                List<Double> resources = new ArrayList<>(3);
+                resources.add(slot.getAllocatedMemOnHeap());
+                resources.add(slot.getAllocatedMemOffHeap());
+                resources.add(slot.getAllocatedCpu());
+                
+                nodePortToResources.put(nodePort, resources);
+            }
+            ret.put(schedEntry.getKey(), nodePortToResources);
+        }
+        return ret;
+    }
+
+    private static Map<String, Map<List<Long>, List<Object>>> computeNewTopoToExecToNodePort(Map<String, SchedulerAssignment> schedAssignments,
+            Map<String, Assignment> existingAssignments) {
+        Map<String, Map<List<Long>, List<Object>>> ret = computeTopoToExecToNodePort(schedAssignments);
+        //Print some useful information
```

Comment:
Minor nit:  space between `//` and `P`?  Seems *most* of the code puts a space after the comment chars.  But egads, 17% of the existing java code in the storm repo doesn't adhere to that common style practice.  (Consistency being a deposed king it seems.)

```
(storm-reviewing) [STORM-1276] % git grep '//\s' | grep ".java:" | wc -l
    4285
(storm-reviewing) [STORM-1276] % git grep '//\S' | grep ".java:" | grep -v -E "htt(p|ps):" | wc -l
     833
```